mina/caip2.md

```diff
@@ -0,0 +1,115 @@
+---
+namespace-identifier: mina-caip2
+title: Mina Namespace - Chains
+author: Mina Foundation
+discussions-to: https://github.com/ChainAgnostic/namespaces/pull/91
+status: Draft
+type: Standard
+created: 2023-09-15
+requires: CAIP-2
+---
+
+# CAIP-2
+
+*For context, see the [CAIP-2][] specification.*
+
+## Abstract
+
+In CAIP-2 a general blockchain identification scheme is defined. This is the implementation of CAIP-2 for Mina network.
+
+### Mina Namespace
+
+The namespace "mina" refers to the wider Mina ecosystem.
+
+## Rationale
+
+Mina consists of multiple networks: a production network `mainnet`, a testing network `testnet` and networks where new features are trialed. Networks trialing new features are identified by a codename, for example `berkeley`. 
+
+An identifier for a Mina chain consists of the namespace prefix "mina:" followed by human readable reference identifying each network. These identifiers may be extended to include codenames and prefixed to include the type of network. These names are separated by "-". 
+
+The type of network will be:
+- devnet, indicating a stable copy of the network used for developer testing.
+- testnet, indicating a less stable copy of the network where new features can be deployed and tested. 
+
+The names chosen above are taken from the signature schema's for each blockchain.
+
+## Syntax
+
+The blockchain namespace and blockchain identifiers will be in lowercase.
+
+The human readable identifier for a network may be one of the following strings:
+
+- `mainnet` 
+- `testnet` 
+- `devnet` 
+- `berkeley`
+
+The human readable identifiers above may be composed to add description using `-` as a separator.
```

Comment:
this is a little confusing -- are `testnet-berkeley` and `testnet` two different networks, or is the former a subset of the nodes of the latter that are running the berkeley nightly/feature-branch of the testnet codebase? i guess i'm confused what "compose" means when discussing network identifiers!

Reply:
ah, ok.  but is this convention already defined somewhere in the code or in the community? seeing that nodes return a `networkID` value implies a convention has already been set somewhere by whomever decides what a valid `networkID` is... is there a normative reference maybe? an MIP or even just a permanent URL in the docs?

Reply:
hi, by "this convention already defined" - do you mean berkeley/devnet/testnet etc or do you mean concatenating berkeley-testnet - if we spun up a test network for the berkeley network? The names are embedded in the code ... the ability to concatenate is intended to "future proof" the convention in case we have more networks, it's new though we can add a definition of this to the docs and link it if it helps ...

Reply:
heh, I guess what I mean by "convention definition" is that nothing *internal* (or native) to the namespace should be defined for the first time here at CASA, or canonically in a CASA spec-- the CASA spec is downstream of mina docs, the only thing being defined here is the export format for cross-chain purposes.  so I guess what I mean is that if there are rules about what future network names will be valid (and instructions on how to embed those in code when you generate a new network/testnet/etc), linking to that from the references would be great!  I highly recommend adding those kinds of definitions to the official docs sooner than later, because you never know who's tinkering a private repo with the expectation of being able to publicly add their private network to the list of known/public chainIds down the road...